<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProbeEditor</class>
 <widget class="QDialog" name="ProbeEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>374</width>
    <height>343</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>编辑探头</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>3</number>
     </property>
     <widget class="QWidget" name="baseInfo">
      <attribute name="title">
       <string>基本信息</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QGroupBox" name="headerBasic">
         <property name="title">
          <string/>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="uuidTitle">
            <property name="text">
             <string>Uuid</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="uuid">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="nameTitle">
            <property name="text">
             <string>名称</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="name"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="manufactureNameTitle">
            <property name="text">
             <string>生产商</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="manufactureName">
            <property name="editable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="modelNameTitle">
            <property name="text">
             <string>型号</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="modelName"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="serialNameTitle">
            <property name="text">
             <string>序列号</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="serialName"/>
          </item>
          <item row="5" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0,0">
            <item>
             <widget class="QDateEdit" name="calibrationDate">
              <property name="dateTime">
               <datetime>
                <hour>0</hour>
                <minute>0</minute>
                <second>0</second>
                <year>2000</year>
                <month>1</month>
                <day>1</day>
               </datetime>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="recoverDate">
              <property name="toolTip">
               <string>更新为今天</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../../resources/icons/Fatcow.qrc">
                <normaloff>:/fatcow/16x16/arrow_rotate_clockwise.png</normaloff>:/fatcow/16x16/arrow_rotate_clockwise.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="deleteDate">
              <property name="toolTip">
               <string>清空</string>
              </property>
              <property name="icon">
               <iconset resource="../../libs/utils/utils.qrc">
                <normaloff>:/utils/images/editclear.png</normaloff>:/utils/images/editclear.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="calibrationDateTitle">
            <property name="text">
             <string>校准时间</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="property">
      <attribute name="title">
       <string>超声</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QGroupBox" name="headerUltrasonic">
         <property name="title">
          <string/>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="typeTitle">
            <property name="text">
             <string>类型</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="probeType">
            <item>
             <property name="text">
              <string>直探头</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>斜探头</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="frequencyTitle">
            <property name="text">
             <string>频率</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="frequency">
            <property name="keyboardTracking">
             <bool>false</bool>
            </property>
            <property name="suffix">
             <string> MHz</string>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="value">
             <double>0.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="focalTypeTitle">
            <property name="text">
             <string>聚焦类型</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QComboBox" name="focalType">
            <item>
             <property name="text">
              <string>点聚焦</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>线聚焦</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>平探头</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="focalTitle">
            <property name="text">
             <string>焦距(水)</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <layout class="QHBoxLayout" name="focalLengthLayout">
            <item>
             <widget class="QDoubleSpinBox" name="focalInch">
              <property name="keyboardTracking">
               <bool>false</bool>
              </property>
              <property name="suffix">
               <string> 英寸</string>
              </property>
              <property name="maximum">
               <double>99999999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="focalSi">
              <property name="suffix">
               <string> mm</string>
              </property>
              <property name="maximum">
               <double>99999999.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="focalZoneLengthTitle">
            <property name="text">
             <string>焦柱长度</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QDoubleSpinBox" name="focalCylinderLength">
            <property name="suffix">
             <string> mm</string>
            </property>
            <property name="maximum">
             <double>99999.990000000005239</double>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="emissionDelayTitle">
            <property name="text">
             <string>发射延迟</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QDoubleSpinBox" name="emissionDelay">
            <property name="suffix">
             <string> μs</string>
            </property>
            <property name="minimum">
             <double>-99999.990000000005239</double>
            </property>
            <property name="maximum">
             <double>99999.990000000005239</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="crystalSizeTitle">
            <property name="text">
             <string>晶片大小</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QDoubleSpinBox" name="sensitivity">
            <property name="suffix">
             <string> dB</string>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="sensitivityTitle">
            <property name="text">
             <string>灵敏度</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="shapeTitle">
            <property name="text">
             <string>形状</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="shape">
            <property name="editable">
             <bool>false</bool>
            </property>
            <property name="currentText">
             <string>方形</string>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <item>
             <property name="text">
              <string>方形</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>圆形</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="4" column="1">
           <layout class="QHBoxLayout" name="focalLengthLayout_2">
            <item>
             <widget class="QDoubleSpinBox" name="crystalSizeInch">
              <property name="keyboardTracking">
               <bool>false</bool>
              </property>
              <property name="suffix">
               <string> 英寸</string>
              </property>
              <property name="maximum">
               <double>99999999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="crystalSizeSi">
              <property name="suffix">
               <string> mm</string>
              </property>
              <property name="maximum">
               <double>99999999.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="geometry">
      <attribute name="title">
       <string>几何位置和尺寸</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QGroupBox" name="headerGeometry">
         <property name="title">
          <string/>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="offsetTitle">
            <property name="text">
             <string>位置偏移</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="offset">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="azimuthTitle">
            <property name="text">
             <string>声束方向角</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="azimuth">
            <property name="keyboardTracking">
             <bool>false</bool>
            </property>
            <property name="suffix">
             <string>°</string>
            </property>
            <property name="minimum">
             <double>-180.000000000000000</double>
            </property>
            <property name="maximum">
             <double>180.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="elevationTitle">
            <property name="text">
             <string>声束仰角</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="elevation">
            <property name="keyboardTracking">
             <bool>false</bool>
            </property>
            <property name="suffix">
             <string>°</string>
            </property>
            <property name="minimum">
             <double>-180.000000000000000</double>
            </property>
            <property name="maximum">
             <double>180.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="lengthTitle">
            <property name="text">
             <string>探头长度</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="length">
            <property name="keyboardTracking">
             <bool>false</bool>
            </property>
            <property name="suffix">
             <string> mm</string>
            </property>
            <property name="minimum">
             <double>-9999.989999999999782</double>
            </property>
            <property name="maximum">
             <double>9999.989999999999782</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="widthTitle">
            <property name="text">
             <string>探头宽度</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QDoubleSpinBox" name="width">
            <property name="suffix">
             <string> mm</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="heightTitle">
            <property name="text">
             <string>探头高度</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QDoubleSpinBox" name="height">
            <property name="suffix">
             <string> mm</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="radiusTitle">
            <property name="text">
             <string>探头半径</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QDoubleSpinBox" name="radius">
            <property name="suffix">
             <string> mm</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="dgs">
      <attribute name="title">
       <string>DGS曲线</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="1" column="4">
        <widget class="QToolButton" name="removeDgs">
         <property name="text">
          <string>-</string>
         </property>
         <property name="icon">
          <iconset resource="../../resources/Resources.qrc">
           <normaloff>:/icons/minus.png</normaloff>:/icons/minus.png</iconset>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QToolButton" name="addDgs">
         <property name="text">
          <string>+</string>
         </property>
         <property name="icon">
          <iconset resource="../../resources/Resources.qrc">
           <normaloff>:/icons/plus.png</normaloff>:/icons/plus.png</iconset>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>259</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="1" colspan="4">
        <widget class="QTreeView" name="dgsTreeView">
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="indentation">
          <number>0</number>
         </property>
         <property name="itemsExpandable">
          <bool>false</bool>
         </property>
         <attribute name="headerDefaultSectionSize">
          <number>80</number>
         </attribute>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="setDefaultTable">
         <property name="text">
          <string>设置为默认</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../libs/utils/utils.qrc"/>
  <include location="../../resources/icons/Fatcow.qrc"/>
  <include location="../../resources/Resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ProbeEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>320</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ProbeEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>320</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
